test/images/image-util.sh

```diff
@@ -117,23 +141,46 @@ push() {
   docker_version_check
   TAG=$(<"${IMAGE}"/VERSION)
   if [[ -f ${IMAGE}/BASEIMAGE ]]; then
-    archs=$(listArchs)
+    base_images=$(listBaseImages)
+    # NOTE(claudiub): if the REMOTE_DOCKER_URL var is not set, or it is an empty string, we must skip
+    # pushing the Windows image and including it into the manifest list.
+    if [[ ! -v "REMOTE_DOCKER_URL" || -z "${REMOTE_DOCKER_URL}" ]]; then
+      if [[ ! -z "$(printf "%s\n" $base_images | grep '^windows')" ]]; then
+        echo "Skipping pushing the image '${IMAGE}' for Windows. REMOTE_DOCKER_URL should be set, containing the URL to a Windows docker daemon."
+        base_images=$(printf "%s\n" $base_images | grep -v "^windows")
+      fi
+    fi
   else
-    archs=${!QEMUARCHS[@]}
+    # prepend linux/ to the QEMUARCHS items.
+    base_images=`printf 'linux/%s\n' "${!QEMUARCHS[@]}"`
   fi
-  for arch in ${archs}; do
-    docker push "${REGISTRY}/${IMAGE}-${arch}:${TAG}"
+  for base_image in ${base_images}; do
+    if [[ $base_image =~ .*/.* ]]; then
+      os_name=`echo $base_image | cut -d "/" -f 1`
+      arch=`echo $base_image | cut -d "/" -f 2`
+    fi
+
+    if [[ "$os_name" = "linux" ]]; then
+      docker push "${REGISTRY}/${IMAGE}:${TAG}-${os_name}-${arch}"
+    else
+      # NOTE(claudiub): We're pushing the image we built on the remote Windows node.
+      docker --tlsverify -H "${REMOTE_DOCKER_URL}" push "${REGISTRY}/${IMAGE}:${TAG}-${os_name}-${arch}"
+    fi
   done
 
   kube::util::ensure-gnu-sed
 
   # The manifest command is still experimental as of Docker 18.09.2
   export DOCKER_CLI_EXPERIMENTAL="enabled"
-  # Make archs list into image manifest. Eg: 'amd64 ppc64le' to '${REGISTRY}/${IMAGE}-amd64:${TAG} ${REGISTRY}/${IMAGE}-ppc64le:${TAG}'
-  manifest=$(echo "$archs" | ${SED} -e "s~[^ ]*~$REGISTRY\/$IMAGE\-&:$TAG~g")
+  # Make base_images list into image manifest. Eg: 'linux/amd64 linux/ppc64le' to '${REGISTRY}/${IMAGE}:${TAG}-linux-amd64 ${REGISTRY}/${IMAGE}:${TAG}-linux-ppc64le'
+  manifest=$(echo "$base_images" | ${SED} "s~\/~-~g" | ${SED} -e "s~[^ ]*~$REGISTRY\/$IMAGE:$TAG\-&~g")
```

Comment:
If the BASEIMAGE files are formatted as

```
os-arch=some/image:tag
```

instead of

```
os/arch=some/image:tag
```

(don't use `/`), does that allow us to simplify the call to to `sed` here (notably, drop the middle one to `${SED} "s~\/~-~g"`? I think because of the `g` it might actually be a bug because you are replacing all instances of `/` which includes the ones possibly embedded in the image names.

Apologies if I am misinterpreting this code.

Reply:
Indeed, I just realized, the names ``listBaseImages`` and ``base_images`` are misleading and they should be renamed. In actuality, the command that is executed is:

    cut -d "=" -f 1 "${IMAGE}"/BASEIMAGE

which basically results in:

    linux/amd64
    linux/arm
    linux/arm64
    linux/ppc64le
    linux/s390x
    windows/amd64

Indeed, we can drop the ``g`` to be safe, but it won't change anything.